components/modal/header.js

```diff
@@ -0,0 +1,31 @@
+import IconButton from '../icon-button';
+import './style.scss';
+import { __ } from '@wordpress/i18n';
+
+const ModalHeader = ( { icon, title, onClose, closeLabel } ) => {
+	const label = closeLabel ? closeLabel : __( 'Close dialog' );
+
+	return (
+		<div
+			className={ 'components-modal__header' }
+		>
+			<div>
+				{ icon &&
+					<span aria-hidden="true">
+						{ icon }
+					</span>
+				}
+				<h1 id="modal-heading" >
```

Comment:
We shouldn't use static IDs, since in a componentized application there's no expectation that there's a maximum of a single instance of a given component. If an ID is necessary, it should be generated as unique via [`withInstanceId`](https://github.com/WordPress/gutenberg/tree/master/components/higher-order/with-instance-id).